### PR TITLE
Add IV, tag, AAD to KeyOperationResult

### PR DESCRIPTION
**Pull request by daviddesberg**
Adds IV, tag, and additional authenticated data to KeyOperationResult. These are necessary for the results of certain symmetric key algorithms and may be necessary for the results of wrap operations in the future. 

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json [hunk 1]
```diff
@@ -1870,6 +1870,24 @@
           "type": "string",
           "format": "base64url",
           "readOnly": true
+        },
+        "iv": {
+          "x-ms-client-name": "result",
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json:1875]: Can't have the same client name (same as below). I'd just elide the property here.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json [hunk 2]
```diff
@@ -1870,6 +1870,24 @@
...
+          "type": "string",
+          "format": "base64url",
+          "readOnly": true
+        },
+        "tag": {
+          "x-ms-client-name": "result",
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json:1881]: I'd call this `authenticatedTag` unless you have a better suggestion. Generated code would at least have the name we plan to go with.

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json [hunk 3]
```diff
@@ -1870,6 +1870,24 @@
...
+          "type": "string",
+          "format": "base64url",
+          "readOnly": true
+        },
+        "aad": {
+          "x-ms-client-name": "result",
```
Comment (heaths) [inline specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json:1887]: I'd call this `additionalAuthenticatedData` unless you have a better suggestion. Generated code would at least have the name we plan to go with.

**Review by heaths — CHANGES_REQUESTED** [review]
You'll probably need to regenerate (or hand-tweak, but it's always good to validate the service does what you expect) the example(s).

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json (hunk 3 above)
Comment (Lusitanian) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json]: fixed

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json (hunk 2 above)
Comment (Lusitanian) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json]: fixed

File: specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json (hunk 1 above)
Comment (Lusitanian) [reply specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/keys.json]: fixed

**Review by heaths — APPROVED** [review]

**Comment by tjprescott** [comment]
@Lusitanian please review the LintDiff failure.

**Comment by Lusitanian** [comment]
@heaths  @tjprescott  this has been sitting here for a while and as I understand from Heath it was important that we get this change in ASAP. The CI usually fails so it's not clear if there's anything in particular I should be looking for?

**Comment by heaths** [comment]
@tjprescott the lint errors I looked at specific have always been there and Key Vault has no plans to fix them at this time. If there's a way to suppress them for now, please advise; otherwise, please merge this.

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
@lmazuel can you merge this? I don't have permissions to merge with the failing LintDiff check.

**Pull request merged**